### PR TITLE
[Refactoring] Remove sdk dependency from DemoApp

### PR DESCRIPTION
**Pull request by pavelprystinka**
## Purpose

* Remove sdk dependency from DemoApp

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by lpeixin — APPROVED** [review]

File: azure-communication-ui/azure-communication-ui/build.gradle [hunk 1]
```diff
@@ -82,7 +82,7 @@ dependencies {
     implementation "androidx.constraintlayout:constraintlayout:$constraint_layout"
     implementation 'androidx.navigation:navigation-fragment-ktx:2.3.5'
     implementation "com.azure.android:azure-core:$azure_core_version"
-    implementation "com.azure.android:azure-communication-calling:$calling_sdk_version"
+    api "com.azure.android:azure-communication-calling:$calling_sdk_version"
```
Comment (ahammer) [inline azure-communication-ui/azure-communication-ui/build.gradle:85]: Do we want this to be API? I'd think we'd want to hide this dependency downstream.

Comment (ahammer) [reply azure-communication-ui/azure-communication-ui/build.gradle:85]: Is there dependencies in the demo for this and can we strip them out?

Comment (pavelprystinka) [reply azure-communication-ui/azure-communication-ui/build.gradle:85]: We need to expose a few types from the SDK related to Authentication

Comment (ahammer) [reply azure-communication-ui/azure-communication-ui/build.gradle:85]: Hmmm, ok makes sense.

**Review by ahammer — APPROVED** [review]

**Pull request merged**